src/bin/client.rs

```diff
@@ -0,0 +1,181 @@
+//! Run this binary with `cargo run --bin client --address <address> --port <port> --name <name>`
+//! or from the binary `./client --address <address> --port <port> --name <name>`
+use clap::Parser;
+use crossterm::{event::*, terminal::*, *};
+use futures::StreamExt;
+use std::io;
+use super_mega_chatroom::*;
+use tokio::net::*;
+use tokio::sync::mpsc;
+use tokio::*;
+use tui::{backend::*, layout::*, widgets::*, *};
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    #[arg(short, long, default_value = "localhost")]
+    address: String,
+    #[arg(short, long, default_value_t = 7878)]
+    port: u16,
+    #[arg(short, long)]
+    name: String,
+}
+
+struct App {
+    name: String,
+    messages: Vec<Message>,
+    input: String,
+    tx: mpsc::Sender<Message>,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args = Args::parse();
+
+    let mut network_stream = TcpStream::connect((args.address.as_str(), args.port)).await?;
+
+    // throw away hi message used for telnet connections to server
+    if let Err(e) = read_from_stream(&mut network_stream).await {
+        eprintln!("Failed to initialize connection to server: {}", e);
+        std::process::exit(1);
+    }
+    println!("Connected to server");
+
+    // Send name to server
+    if let Err(e) = write_to_stream(&mut network_stream, format!("{}\n", args.name)).await {
+        eprintln!("Failed to initialize connection with server: {}", e);
+        std::process::exit(1);
+    }
+
+    let (tx, mut rx) = mpsc::channel(64);
+
+    let mut app = App {
+        name: args.name,
+        messages: vec![],
+        input: String::new(),
+        tx,
+    };
+
+    let mut terminal = prepare_terminal()?;
+    let mut event_stream = EventStream::new();
+
+    loop {
+        select! {
+            msg = read_from_stream(&mut network_stream) => {
+                match msg {
+                    Ok(msg) => {
+                        let x = msg.split(':').collect::<Vec<&str>>();
+                        let name = x[0].trim().to_string();
+                        let msg = x[1].trim().to_string();
+                        app.messages.push(Message { name, msg, id: 0 })
```

Comment:
id here being 0 always seems incorrect

Reply:
Doesn't matter, cause this is on the client. Client doesn't care about id at all. Really i should've not used the Message struct in client at all but was late and i was tired lol